types/net_fields.d.ts

```diff
@@ -0,0 +1,321 @@
+export type Fields = {
+};
+
+export type NetFields = Fields & {
+  event: 0;
+  timestamp: 1;
+};
+
+// 0x00
+export type NetGameLogFields = NetFields & {
+  code: 2;
+  name: 3;
+  line: 4;
+};
+
+// 0x01
+export type NetChangeZoneFields = NetFields & {
+  id: 2;
+  name: 3;
+};
+
+// 0x02, copied from emulator
+export type NetChangedPlayerFields = NetFields & {
+  id: 2;
+  name: 3;
+};
+
+// 0x03
+export type NetAddedCombatantFields = NetFields & {
+  id: 2;
+  name: 3;
+  job: 4;
+  level: 5;
+  ownerId: 6;
+  worldId: 7;
+  world: 8;
+  npcNameId: 9;
+  npcBaseId: 10;
+  currentHp: 11;
+  hp: 12;
+  // currentMp: 13;
+  // mp: 14;
+  // currentTp: 15;
+  // tp: 16;
+  x: 17;
+  y: 18;
+  z: 19;
+  heading: 20;
+};
+
+// 0x04, identical to 0x03 for fields
+export type NetRemovedCombatantFields = NetAddedCombatantFields;
+
+// 0x0C
+export type NetPlayerStatsFields = NetFields & {
+  job: 2;
+  strength: 3;
+  dexterity: 4;
+  vitality: 5;
+  intelligence: 6;
+  mind: 7;
+  piety: 8;
+  attackPower: 9;
+  directHit: 10;
+  criticalHit: 11;
+  attackMagicPotency: 12;
+  healMagicPotency: 13;
+  determination: 14;
+  skillSpeed: 15;
+  spellSpeed: 16;
+  tenacity: 18;
+};
+
+// 0x0D, non-net-log only
+export type NetHasHPFields = NetFields & {
+  name: 2;
+  hp: 3;
+};
```

Comment:
Would it make sense to just drop this? See comment in `trigger.d.ts` about having triggers only be netregex.